src/Constraint/Bound.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+
+namespace Composer\Semver\Constraint;
+
+
+class Bound
+{
+    /**
+     * @var string
+     */
+    private $version;
+
+    /**
+     * @var bool
+     */
+    private $isInclusive;
+
+    /**
+     * @param string $version
+     * @param bool $isInclusive
+     */
+    public function __construct($version, $isInclusive)
+    {
+        $this->version = $version;
+        $this->isInclusive = $isInclusive;
+    }
+
+    /**
+     * @return string
+     */
+    public function getVersion()
+    {
+        return $this->version;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInclusive()
+    {
+        return $this->isInclusive;
+    }
+
+    /**
+     * Compares a bound to another with a given operator
+     *
+     * @param Bound $other
+     * @param string $operator
+     *
+     * @return bool
+     */
+    public function compareTo(Bound $other, $operator)
+    {
+        if (!in_array($operator, array('<', '>'))) {
+            throw new \InvalidArgumentException('Does not support any other operator other than > or <.');
+        }
+
+        // If they are the same it doesn't matter
+        if ($this == $other) {
+            return false;
+        }
+
+        // Not the same version means we can use regular version comparison and inclusive/exclusive doesn't matter
+        if (!version_compare($this->getVersion(), $other->getVersion(), '==')) {
```

Comment:
Could avoid running version compare twice here, by just running it once without an operator and using the result variable?

Reply:
I need to know two results. 1. if they are equal and 2. if the match the given operator. So it will always be two operators = two comparisons.

Reply:
No need, you can just do version_compare and then check if negative, positive or 0 and compare to the operator?

Reply:
Ah, I misunderstood this one. You're right. Fixed in 83cc74a7de4008993754ebb591fc6cb5f1d7344d.